### DISCUSSION TITLE
Implementing Extracellular Stimulus Pulses with Different Magnitudes

### DISCUSSION THREAD

**Discussion by joyce-alysia** [Chaste/Chaste#141] [Q&A]
Hi all, I am trying to use a bath simulation with an extracellular stimulus to administer two electrode pulses at different times and magnitudes (i.e. an S1-S2 simulation, where S1 does not equal S2). I got the different timing working by adjusting the conditionals in the `SwitchOn()` and `SwitchOff()` functions in the `Electrodes` class, but I am finding it more challenging to get different magnitudes for each of the pulses. If I am correct, it looks like Chaste normally evaluates the stimulus Neumann boundary conditions and stores them at the instantiation of the electrodes class in a binary way such that `SwitchOn()` triggers one stimulus boundary condition and `SwitchOff()` triggers the default homogeneous BCs. This happens before the time loop, but I am trying to get a more dynamic evaluation of the boundary conditions. 

The one thing I tried that "almost" succeeded was passing time to the `Electrodes()` constructor, so that `input_flux` and `output_flux` could be conditionally changed, and then calling `SetElectrodes()` every time `SwitchOn()` is true (i.e. in `BidomainProblem<DIM>::AtBeginningOfTimestep()` ). 

This worked in that there were two pulses of different magnitudes at the right times, but the action potentials produced were unusual--the first (set to above threshold) produced graded potential spikes where some nodes spiked higher than normal and others spiked less than 0, and the S2 pulse produced a normal action potential, regardless of what the S2 magnitude was set to. 

I have gone through with a debugger and tracked the boundary conditions being passed to `AddNeumannBoundaryCondition()`, and those are being assigned as expected at the proper times. Does anyone have any ideas about what might be happening here or what would be the best way to implement this type of simulation? Thanks in advance!

**Comment by mirams** [comment]
Hi, we are having a look at this, sorry it has taken a while, and @jmpf or I will get back to you soon, do you have a short 'failing test' you could supply to show the problem?

**Reply by jmpf** [reply]
Apologies from me too.  I don't think any of the current developers have been looking at the code for the electrode stimuli for a very long time---but we don't mind getting back up to speed.

**Comment by fcooper8472** [comment]
Closing this as no failing further information has been provided.

@joyce-alysia please feel free to re-open if you need any more help!

**Comment by mirams** [comment]
I'll keep it open to remember to look into it, I think it is a 'sensible question' that people might want to search and look back at
